src/anemoi/inference/outputs/netcdf.py

```diff
@@ -36,22 +37,23 @@ class NetCDFOutput(Output):
     def __init__(
         self,
         context: Context,
-        path: str,
+        path: Path,
```

Comment:
Technically it's still a string when it's instantiated from a config, I guess the type hint should be `str | Path` ?

There's also a little duplication going on with the casting to `Path` and creating the parent directories in all the different outputs. Would a new decorator be a good idea, something like `@ensure_path("path")` ?

But also happy to leave it like this.